workers/unity/Packages/com.improbable.gdk.deploymentlauncher/.DeploymentLauncher/Program.cs

```diff
@@ -0,0 +1,323 @@
+                simDeploymentSnapshotFilePath = args[8];
+            }
+
+            // Create service clients.
+            var playerAuthServiceClient = PlayerAuthServiceClient.Create();
+            var snapshotServiceClient = SnapshotServiceClient.Create();
+            var deploymentServiceClient = DeploymentServiceClient.Create();
+
+            try
+            {
+                // Upload snapshots.
+                var mainSnapshotId = UploadSnapshot(snapshotServiceClient, mainDeploymentSnapshotFilePath, projectName,
+                    mainDeploymentName);
+                if (mainSnapshotId.Length == 0)
+                {
+                    return 1;
+                }
+
+                var simSnapshotId = "";
+                if (launchSimPlayerDeployment)
+                {
+                    simSnapshotId = UploadSnapshot(snapshotServiceClient, simDeploymentSnapshotFilePath, projectName,
+                        simDeploymentName);
+                    if (simSnapshotId.Length == 0)
+                    {
+                        return 1;
+                    }
+                }
+
+                // Create main deployment.
+                var mainDeploymentConfig = new Deployment
+                {
+                    AssemblyId = assemblyName,
+                    LaunchConfig = new LaunchConfig
+                    {
+                        ConfigJson = File.ReadAllText(mainDeploymentJson)
+                    },
+                    Name = mainDeploymentName,
+                    ProjectName = projectName,
+                    StartingSnapshotId = mainSnapshotId
+                };
+                if (launchSimPlayerDeployment)
+                {
+                    // This tag needs to be added to allow simulated clients to connect using login
+                    // tokens generated with anonymous auth.
+                    mainDeploymentConfig.Tag.Add("dev_login");
+                }
+
+                Console.WriteLine(
+                    $"Creating the main deployment {mainDeploymentName} in project {projectName} with snapshot ID {mainSnapshotId}.");
+                var mainDeploymentCreateOp = deploymentServiceClient.CreateDeployment(new CreateDeploymentRequest
+                {
+                    Deployment = mainDeploymentConfig
+                }).PollUntilCompleted();
+                Console.WriteLine("Successfully created the main deployment.");
+
+                if (launchSimPlayerDeployment)
+                {
+                    // Create development authentication token used by the simulated players.
+                    var dat = playerAuthServiceClient.CreateDevelopmentAuthenticationToken(
+                        new CreateDevelopmentAuthenticationTokenRequest
+                        {
+                            Description = "DAT for sim worker deployment.",
+                            Lifetime = Duration.FromTimeSpan(new TimeSpan(7, 0, 0, 0)),
```

Comment:
for how long do we need the dat? would prefer to make the timespan as short as possible

Reply:
I have no idea. cc @davedissian ?

Reply:
Doesn't actually matter. I figured the DAT should last for as long as the deployment is running for, right? (otherwise sim workers wont be able to connect anymore).

Reply:
in that case we should change how we define it / make it possible to regenerate it as setting it to one week will not give us that guarantee and we might end up with generating multiple of those . But fine for now :)